src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -191,11 +202,79 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
   }
 
   /**
-   * Toggles the view of spreadsheet interface
+   * Toggles the view of 'New Students' spreadsheet interface
    * and/or its affiliated buttons
    */
-  togglePanel(): void {
-    this.isCollapsed = !this.isCollapsed;
+  toggleNewStudentsPanel(): void {
+    this.isNewStudentsPanelCollapsed = !this.isNewStudentsPanelCollapsed;
+  }
+
+  /**
+   * Returns the length of the current spreadsheet.
+   * Rows with all null values are filtered.
+   */
+  getSpreadsheetLength(dataHandsontable: string[][]): number {
+    return dataHandsontable
+        .filter((row: string[]) => (!row.every((cell: string) => cell === null)))
+        .length;
+  }
+
+  /**
+   * Transforms the first uppercase letter of a string into a lowercase letter.
+   */
+  unCapitalizeFirstLetter(targetString: string): string {
+    return targetString.charAt(0).toLowerCase() + targetString.slice(1);
+  }
+
+  /**
+   * Converts returned student list to a suitable format required by Handsontable.
+   */
+  studentListDataToHandsontableData(studentsData: StudentAttributes[], handsontableColHeader: any[]): string[][] {
+    const headers: string[] = handsontableColHeader.map(this.unCapitalizeFirstLetter);
+    return studentsData.map((student: StudentAttributes) => (headers.map(
+        (header: string) => (student as any)[header])));
+  }
+
+  /**
+   * Loads existing student data into the spreadsheet interface.
+   */
+  loadExistingStudentsData(existingStudentsHOTInstance: Handsontable, studentsData: StudentAttributes[]): void {
+    existingStudentsHOTInstance.loadData(this.studentListDataToHandsontableData(
+        studentsData, (existingStudentsHOTInstance.getColHeader() as any[])));
+  }
+
+  /**
+   * Toggles the view of 'Existing Students' spreadsheet interface
+   */
+  toggleExistingStudentsPanel(): void {
+    this.isExistingStudentsPanelCollapsed = !this.isExistingStudentsPanelCollapsed;
+    this.loading = true;
+    const existingStudentsHOTInstance: Handsontable =
+        this.hotRegisterer.getInstance(this.existingStudentsHOT);
+
+    // Calling REST API only the first time when spreadsheet has no data
+    if (this.getSpreadsheetLength(existingStudentsHOTInstance.getData()) !== 0) {
+      this.loading = false;
+      return;
+    }
+
+    const paramMap: { [key: string]: string } = {
+      courseid: this.courseid,
```

Comment:
Add `user` to enable masquerade mode.

Reply:
I realized `/course/enroll/students` didn't have `user` passed in as a param as well. Added that in as well.